torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5461,10 +5461,17 @@ def random_index(shape):
              ((M, 1, M), (), (M, M, 1), True),
              ((), (M, M), (), True),)
 
+    cases_scalar = (((S, S), (S, S), 0.5, False),
```

Comment:
At least one of these cases should probably be scalar x tensor and scalar x scalar, right?

Can we include a complex scalar here, too?

Reply:
Currently `input` has to be Tensor or a list of Tensor, thus we are checking only tensor x scalar case.

Other cases are tested in `test_torch`

Reply:
Isn't the "input" tensor the boolean condition tensor, though?

Reply:
https://github.com/pytorch/pytorch/blob/4176afc4a035c0e79281b4955772c8962bc3ff4f/torch/testing/_internal/common_methods_invocations.py#L9651-L9655

Reply:
Aha, tricky